code/src/java/plugin/primitive/pobject/AbilityToken.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2010-15 (C) Thomas Parker <thpr@users.sourceforge.net>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+package plugin.primitive.pobject;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+
+import pcgen.cdom.base.Category;
+import pcgen.cdom.base.Converter;
+import pcgen.cdom.content.CNAbility;
+import pcgen.cdom.enumeration.GroupingState;
+import pcgen.cdom.reference.CDOMSingleRef;
+import pcgen.core.Ability;
+import pcgen.core.AbilityCategory;
+import pcgen.core.PlayerCharacter;
+import pcgen.rules.context.LoadContext;
+import pcgen.rules.persistence.token.PrimitiveToken;
+import pcgen.util.Logging;
+
+public class AbilityToken<T> implements PrimitiveToken<T>
```

Comment:
This implements CHOOSE:ABILITY rather than ABILITY right? That might be good as a class comment.